vrx_gz/src/ScanDockScoringPlugin.cc

```diff
@@ -0,0 +1,311 @@
+  /// \param[in] _expectedColors The sequence of expected colors.
+  /// \param[in] _colorSequenceTopic The topic used to receive the color
+  /// submisison.
+  public: ColorSequenceChecker(const std::vector<std::string> &_expectedColors,
+                               const std::string &_colorSequenceTopic);
+  /// \brief Enable submissions.
+  public: void Enable();
+
+  /// \brief Disable submissions.
+  public: void Disable();
+
+  /// \brief Whether a team submitted the color sequence or not.
+  /// \return True when the submission was received or false otherwise.
+  public: bool SubmissionReceived() const;
+
+  /// \brief Whether a team submitted the color sequence and is correct or not.
+  /// \return True when the team submitted the color sequence and it is correct
+  /// or false otherwise.
+  public: bool Correct() const;
+
+  /// \brief Callback executed when a new color submission is received.
+  /// \param[in] _sequence Contains the submission. The size of the vector
+  /// should be three.
+  private: void OnColorSequence(const msgs::StringMsg_V &_sequence);
+
+  /// \brief The expected color sequence.
+  private: std::vector<std::string> expectedSequence;
+
+  /// \brief Topic where the color sequence should be sent.
+  private: std::string colorSequenceTopic;
+
+  /// \brief The transport node.
+  private: transport::Node node;
+
+  /// \brief Whether the color sequence has been received or not.
+  private: bool colorSequenceReceived = false;
+
+  /// \brief Whether the color sequence received is correct or not.
+  private: bool correctSequence = false;
+};
+
+//////////////////////////////////////////////////
+ColorSequenceChecker::ColorSequenceChecker(
+  const std::vector<std::string> &_expectedColors,
+  const std::string &_colorSequenceTopic)
+  : expectedSequence(_expectedColors),
+    colorSequenceTopic(_colorSequenceTopic)
+{
+}
+
+//////////////////////////////////////////////////
+void ColorSequenceChecker::Enable()
+{
+  this->node.Subscribe(this->colorSequenceTopic,
+    &ColorSequenceChecker::OnColorSequence, this);
+}
+
+//////////////////////////////////////////////////
+void ColorSequenceChecker::Disable()
+{
+  this->node.Unsubscribe(this->colorSequenceTopic);
+}
+
+/////////////////////////////////////////////////
+bool ColorSequenceChecker::SubmissionReceived() const
+{
+  return this->colorSequenceReceived;
+}
+
+/////////////////////////////////////////////////
+bool ColorSequenceChecker::Correct() const
+{
+  return this->correctSequence;
+}
+
+/////////////////////////////////////////////////
+void ColorSequenceChecker::OnColorSequence(const msgs::StringMsg_V &_sequence)
+{
+  gzmsg << "ColorSequenceChecker: Color sequence submission received:"
+        << _sequence.DebugString() << std::endl;
+  std::cout << std::flush;
+
+  // Sanity check: Only one submission is allowed.
+  if (this->colorSequenceReceived)
+  {
+    gzerr << "The color sequence has already been submitted" << std::endl;
```

Comment:
This does not seem to be displaying, despite the fact that the variable gets set to true in line 137, below. Maybe the problem is that the color checker gets disabled and stops listening for messages.

Reply:
It's a bit redundant because as you said, it's disabled when the first submission is processed.